grafana-plugin/src/pages/incidents/Incidents.tsx

```diff
@@ -423,27 +448,46 @@ class Incidents extends React.Component<IncidentsPageProps, IncidentsPageState>
                 ? `${selectedIncidentIds.length} Alert Group${selectedIncidentIds.length > 1 ? 's' : ''} selected`
                 : 'No Alert Groups selected'}
             </Text>
-          </HorizontalGroup>
-        </div>
-        {hasInvalidatedAlert && (
-          <div className={cx('out-of-date')}>
-            <Text type="secondary">Results out of date</Text>
-            <Button
-              style={{ marginLeft: '8px' }}
-              disabled={store.alertGroupStore.alertGroupsLoading}
-              variant="primary"
-              onClick={this.onIncidentsUpdateClick}
-            >
-              Refresh
-            </Button>
           </div>
-        )}
+
+          <div className={cx('fields-dropdown')}>
+            {hasInvalidatedAlert && (
+              <HorizontalGroup spacing="xs">
+                <Text type="secondary">Results out of date</Text>
+                <Button
+                  style={{ marginLeft: '8px' }}
+                  disabled={store.alertGroupStore.alertGroupsLoading}
+                  variant="primary"
+                  onClick={this.onIncidentsUpdateClick}
+                >
+                  Refresh
+                </Button>
+              </HorizontalGroup>
+            )}
+
+            {store.hasFeature(AppFeature.Labels) && (
+              <RadioButtonGroup
+                options={[
+                  { value: false, icon: 'wrap-text' },
+                  {
+                    value: true,
+                    icon: 'arrow-from-right',
+                  },
+                ]}
```

Comment:
Doesn't matter in one case but doing that across the codebase matters. In general if we have value of nonprimitive like object/array that doesn't depend on anything from within the component body, it's good to move it outside the render. Otherwise the array is recreated on each render and can cause additional unneeded rerenders of the child